src/components/menu/menu-full-screen/menu-full-screen.component.js

```diff
@@ -33,13 +44,30 @@ const MenuFullscreen = ({
     }
   };
 
-  useLayoutEffect(() => {
-    const checkTransitionEnd = () => {
-      menuContentRef.current.focus();
+  useEffect(() => {
```

Comment:
I'm a little confused by this implementation. The `useLayoutEffect` will run before the `useEffect`. So as far as I can tell this will run `setAnimationComplete(true)`, then it will `setAnimationComplete(false)` then `setAnimationComplete(true)`


I also noticed it doesn't seem to reset the state as expected so the trap only works when the menu is opened the first time etc
https://user-images.githubusercontent.com/44157880/167400283-dc0fd652-bf48-47e2-8d6a-fed061fd3507.mov

Reply:
Interesting, thanks - I'll look into this. The useLayoutEffect is just because this was using useLayoutEffect before, even though I've changed the internals a little - I never dived into why it needed to wait for the actual DOM to be updated (I think that's the only difference between useLayoutEffect and useEffect?).

Either way your demo clearly shows a serious bug that I'd completely missed so I will look into this!